diff --git a/tabbed.c b/tabbed.c
index eafe28a..aaea165 100644
--- a/tabbed.c
+++ b/tabbed.c
@@ -1000,10 +1000,11 @@
 	while (waitpid(-1, NULL, WNOHANG) > 0);
 
 	/* init screen */
+	int tabHeight;
 	screen = DefaultScreen(dpy);
 	root = RootWindow(dpy, screen);
 	initfont(font);
-	bh = dc.h = dc.font.height + 2;
+	bh = dc.h = tabHeight;
 
 	/* init atoms */
 	wmatom[WMDelete] = XInternAtom(dpy, "WM_DELETE_WINDOW", False);
