From d44024135b691f7987c886edcdf76c22fa7acbf5 Mon Sep 17 00:00:00 2001
From: appppless <1260319163@qq.com>
Date: Mon, 29 Dec 2025 19:14:39 +0800
Subject: [PATCH] add check for --remove-control-file for bittorrent download

---
 src/RequestGroup.cc | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/RequestGroup.cc b/src/RequestGroup.cc
index fd6801fcc4..1d018b4330 100644
--- a/src/RequestGroup.cc
+++ b/src/RequestGroup.cc
@@ -407,6 +407,15 @@ void RequestGroup::createInitialCommand(
     }
 
     removeDefunctControlFile(progressInfoFile);
+    // Check if user requested to remove control file
+    if (!option_->getAsBool(PREF_DRY_RUN) &&
+        option_->getAsBool(PREF_REMOVE_CONTROL_FILE) &&
+        progressInfoFile->exists()) {
+      progressInfoFile->removeFile();
+      A2_LOG_NOTICE(fmt(_("Removed control file for %s because it is requested by"
+                          " user."),
+                        progressInfoFile->getFilename().c_str()));
+    }
     {
       int64_t actualFileSize = pieceStorage_->getDiskAdaptor()->size();
       if (actualFileSize == downloadContext_->getTotalLength()) {
@@ -431,7 +440,8 @@ void RequestGroup::createInitialCommand(
     else if (pieceStorage_->getDiskAdaptor()->fileExists()) {
       if (!option_->getAsBool(PREF_CHECK_INTEGRITY) &&
           !option_->getAsBool(PREF_ALLOW_OVERWRITE) &&
-          !option_->getAsBool(PREF_BT_SEED_UNVERIFIED)) {
+          !option_->getAsBool(PREF_BT_SEED_UNVERIFIED) &&
+          !option_->getAsBool(PREF_REMOVE_CONTROL_FILE)) {
         // TODO we need this->haltRequested = true?
         throw DOWNLOAD_FAILURE_EXCEPTION2(
             fmt(MSG_FILE_ALREADY_EXISTS,
